modules/create_timepoint/php/create_timepoint.class.inc

```diff
@@ -34,290 +33,27 @@ class Create_Timepoint extends \NDB_Form
     protected $candidate;
 
     /**
-     * The subprojectID being created
-     */
-    var $subprojectID;
-
-    /**
-     * Check user permissions
+     * Determine whether the user has permission to view this page
      *
      * @param \User $user The user whose access is being checked
      *
      * @return bool
      */
     function _hasAccess(\User $user) : bool
     {
-        $candidate = \Candidate::singleton(new CandID($this->identifier));
+        $this->candidate = \Candidate::singleton(new CandID($this->identifier));
 
         // check user permissions
         return (
             $user->hasPermission('data_entry') &&
             (in_array(
-                $candidate->getData('RegistrationCenterID'),
+                $this->candidate->getData('RegistrationCenterID'),
```

Comment:
i can't seem to be able to do a multi-line suggestion but I think the in_array() check from RegistrationcenterID and centerIDs can be replaced with `$candidate->isAccessibleBy($user)` as was done in this [PR](https://github.com/aces/Loris/pull/6093) for timepoint_list

Reply:
Thanks @zaliqarosli, I didn't know about that method. Also it can replace the permission check for loading the module as well. Cool

Reply:
yeah it's pretty neat! could you still point to me to what you did to make the 'data_entry' permission actually work?

Reply:
@zaliqarosli oops, checkout the `_verifyCreatePermissions` function inside the timepoint.class.inc on line 221.